## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :x: [more...](#3af16f6b-8bc5-48ce-a62b-90520b52e18f) |
| mint-fs.sh | :x: [more...](#dd9b3d43-d98e-4033-a49f-7189aa963688) |
| mint-gateway-s3.sh | :x: [more...](#16aa88ea-c3d1-4351-a86e-bb93c5d5f27e) |
| mint-gateway-azure.sh | :x: [more...](#16768376-ee51-4076-996f-956597f1b58b) |
| mint-erasure.sh | :x: [more...](#986de919-fe51-40f3-a04e-4368ec4f6007) |
| mint-dist-erasure.sh | :x: [more...](#fd680a07-33d6-47b2-8cd8-1498831df0f1) |
| mint-zoned.sh | :x: [more...](#2a594828-a5f0-4824-8bf3-555e4ec4ae22) |
| mint-gateway-nas.sh | :x: [more...](#31307198-9c94-4224-8d0f-a09683c4e351) |
<a name="fd680a07-33d6-47b2-8cd8-1498831df0f1"></a>
### 11136-55d53a2/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:30096
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a92460386da3:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-efotort3hu1hyg",
    "objectName": "ws2nrlcvdxbc5z00iqlgsnql0cqrq0"
  },
  "duration": 280,
  "error": "InvalidPart: One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\n\tstatus code: 400, request id: 1651FE9279736E73, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 1 seconds
(2/15) Running aws-sdk-php tests ... done in 43 seconds
(3/15) Running aws-sdk-ruby tests ... done in 4 seconds
(4/15) Running awscli tests ... done in 1 minutes and 17 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 38 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 0 seconds
(8/15) Running minio-go tests ... done in 2 minutes and 27 seconds
(9/15) Running minio-java tests ... done in 1 minutes and 25 seconds
(10/15) Running minio-js tests ... done in 58 seconds
(11/15) Running minio-py tests ... done in 3 minutes and 0 seconds
(12/15) Running s3cmd tests ... done in 21 seconds
(13/15) Running s3select tests ... FAILED in 1 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="2a594828-a5f0-4824-8bf3-555e4ec4ae22"></a>
### 11136-55d53a2/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32740
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5914945a9487:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-nwuitb3rof5nz6",
    "objectName": "l59v6jwxhs4hilvwoxr4kqccgln6d9"
  },
  "duration": 319,
  "error": "InvalidPart: One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\n\tstatus code: 400, request id: 1651FE929A5E62E0, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 2 seconds
(2/15) Running aws-sdk-php tests ... done in 43 seconds
(3/15) Running aws-sdk-ruby tests ... done in 4 seconds
(4/15) Running awscli tests ... done in 2 minutes and 17 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 47 seconds
(7/15) Running minio-dotnet tests ... done in 51 seconds
(8/15) Running minio-go tests ... done in 2 minutes and 6 seconds
(9/15) Running minio-java tests ... done in 1 minutes and 35 seconds
(10/15) Running minio-js tests ... done in 57 seconds
(11/15) Running minio-py tests ... done in 3 minutes and 29 seconds
(12/15) Running s3cmd tests ... done in 22 seconds
(13/15) Running s3select tests ... FAILED in 0 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="31307198-9c94-4224-8d0f-a09683c4e351"></a>
### 11136-55d53a2/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:32231
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f2def66412cd:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-0uxo9ef0mzwtbs",
    "objectName": "v6xazkynzo4e4othzutt2xe6e5jcxg"
  },
  "duration": 20,
  "error": "InvalidPart: One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\n\tstatus code: 400, request id: 1651FE9258EC7BB0, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 1 seconds
(2/15) Running aws-sdk-php tests ... done in 41 seconds
(3/15) Running aws-sdk-ruby tests ... done in 2 seconds
(4/15) Running awscli tests ... done in 2 minutes and 1 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 40 seconds
(7/15) Running minio-dotnet tests ... done in 36 seconds
(8/15) Running minio-go tests ... done in 1 minutes and 6 seconds
(9/15) Running minio-java tests ... done in 19 seconds
(10/15) Running minio-js tests ... done in 32 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 12 seconds
(12/15) Running s3cmd tests ... done in 13 seconds
(13/15) Running s3select tests ... FAILED in 1 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="986de919-fe51-40f3-a04e-4368ec4f6007"></a>
### 11136-55d53a2/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30085
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 58a0e213c5c3:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6il36zlw1bh3ic",
    "objectName": "gm2xsc5rhdawh4tee2yp2qb2bdfdgh"
  },
  "duration": 167,
  "error": "InvalidPart: One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\n\tstatus code: 400, request id: 1651FE926CB098EA, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 2 seconds
(2/15) Running aws-sdk-php tests ... done in 42 seconds
(3/15) Running aws-sdk-ruby tests ... done in 2 seconds
(4/15) Running awscli tests ... done in 1 minutes and 11 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 26 seconds
(7/15) Running minio-dotnet tests ... done in 39 seconds
(8/15) Running minio-go tests ... done in 1 minutes and 25 seconds
(9/15) Running minio-java tests ... done in 45 seconds
(10/15) Running minio-js tests ... done in 46 seconds
(11/15) Running minio-py tests ... done in 2 minutes and 21 seconds
(12/15) Running s3cmd tests ... done in 16 seconds
(13/15) Running s3select tests ... FAILED in 1 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="16aa88ea-c3d1-4351-a86e-bb93c5d5f27e"></a>
### 11136-55d53a2/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:32460
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 390a11383a91:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-tewoeo0zyhgjjs",
    "objectName": "kzlh6qstoxvmci1cx6ma4ud45lnovb"
  },
  "duration": 42,
  "error": "InvalidPart: One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\n\tstatus code: 400, request id: 1651FE926E25D232, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 2 seconds
(2/15) Running aws-sdk-php tests ... done in 41 seconds
(3/15) Running aws-sdk-ruby tests ... done in 2 seconds
(4/15) Running awscli tests ... done in 2 minutes and 0 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 44 seconds
(7/15) Running minio-dotnet tests ... done in 31 seconds
(8/15) Running minio-go tests ... done in 44 seconds
(9/15) Running minio-java tests ... done in 20 seconds
(10/15) Running minio-js tests ... done in 35 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 12 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 6 seconds
(13/15) Running s3select tests ... FAILED in 0 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="16768376-ee51-4076-996f-956597f1b58b"></a>
### 11136-55d53a2/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:32285
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9237f598e0c7:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 7 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-t5zz0cvdt0tj51",
    "objectName": "k096irvxo9ullf2t1nymvhmtepej4e"
  },
  "duration": 2646,
  "error": "NoSuchKey: The specified key does not exist.\n\tstatus code: 404, request id: 1651FE93BBAF3108, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 2 seconds
(2/15) Running aws-sdk-php tests ... done in 2 minutes and 50 seconds
(3/15) Running aws-sdk-ruby tests ... done in 21 seconds
(4/15) Running awscli tests ... done in 2 minutes and 50 seconds
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 3 minutes and 28 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 43 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 49 seconds
{
  "args": {
    "bucketName": "minio-go-test-ifqw6tya2td01zd3",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 1186,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 9 minutes and 16 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[user metadata]",
  "duration": 175,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = AuthenticationFailed, message = -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====\nDescription=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:7cdaf286-101e-00cd-23b5-d588a6000000\nTime:2020-12-19T03:16:32.5746451Z, Details: \n   AuthenticationErrorDetail: The MAC signature found in the HTTP request 'LhrL3V8HJW75Tfpuofa1cUlP2pqZbZs5EKHV2nhWeSs=' is not the same as any computed signature. Server used following string to sign: 'PUT\n\n\n128\n\napplication/xml\n\n\n\n\n\n\nx-ms-blob-cache-control:\nx-ms-blob-content-disposition:\nx-ms-blob-content-encoding:\nx-ms-blob-content-language:\nx-ms-blob-content-type:application/octet-stream\nx-ms-client-request-id:ff29dd04-86e2-4132-67ec-a545fb35837e\nx-ms-date:Sat, 19 Dec 2020 03:16:32 GMT\nx-ms-meta-my_header1:a   b   c\nx-ms-meta-my_header2:\"a   b   c\"\nx-ms-meta-my_project:Project One\nx-ms-meta-my_unicode_tag:å•†å“\nx-ms-version:2019-02-02\n/minioazure/minio-java-test-3mdvjhg/minio-java-test-2vdu5c4\ncomp:blocklist\ntimeout:1501'.\n   Code: AuthenticationFailed\n   PUT https://minioazure.blob.core.windows.net/minio-java-test-3mdvjhg/minio-java-test-2vdu5c4?comp=blocklist&timeout=1501\n   Authorization: REDACTED\n   Content-Length: [128]\n   Content-Type: [application/xml]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2020-12-19T02:47:46Z]\n   X-Ms-Blob-Cache-Control: []\n   X-Ms-Blob-Content-Disposition: []\n   X-Ms-Blob-Content-Encoding: []\n   X-Ms-Blob-Content-Language: []\n   X-Ms-Blob-Content-Type: [application/octet-stream]\n   X-Ms-Client-Request-Id: [ff29dd04-86e2-4132-67ec-a545fb35837e]\n   X-Ms-Date: [Sat, 19 Dec 2020 03:16:32 GMT]\n   X-Ms-Meta-My_header1: [a   b   c]\n   X-Ms-Meta-My_header2: [\"a   b   c\"]\n   X-Ms-Meta-My_project: [Project One]\n   X-Ms-Meta-My_unicode_tag: [商品]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\n   Content-Length: [1092]\n   Content-Type: [application/xml]\n   Date: [Sat, 19 Dec 2020 03:16:32 GMT]\n   Server: [Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [AuthenticationFailed]\n   X-Ms-Request-Id: [7cdaf286-101e-00cd-23b5-d588a6000000]\n\n\n, bucketName = minio-java-test-3mdvjhg, objectName = minio-java-test-2vdu5c4, resource = /minio-java-test-3mdvjhg/minio-java-test-2vdu5c4, requestId = 1651FFCBBCF14FFC, hostId = 8bbc7aca-ec52-4bcb-920e-81438ed8a651)\nrequest={method=PUT, url=http://minio-dev8.minio.io:32285/minio-java-test-3mdvjhg/minio-java-test-2vdu5c4, headers=x-amz-meta-My-Unicode-Tag: 商品\nx-amz-meta-My-Project: Project One\nx-amz-meta-My-header1: a   b   c\nx-amz-meta-My-Header2: \"a   b   c\"\nContent-Type: application/octet-stream\nHost: minio-dev8.minio.io:32285\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20201219T031632Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20201219/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-meta-my-header1;x-amz-meta-my-header2;x-amz-meta-my-project;x-amz-meta-my-unicode-tag, Signature=*REDACTED*\n}\nresponse={code=403, headers=Accept-Ranges: bytes\nContent-Length: 3086\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1651FFCBBCF14FFC\nX-Xss-Protection: 1; mode=block\nDate: Sat, 19 Dec 2020 03:16:32 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.putObject(MinioClient.java:4547), io.minio.MinioClient.putObject(MinioClient.java:2713), io.minio.MinioClient.putObject(MinioClient.java:2830), FunctionalTest.testPutObject(FunctionalTest.java:763), FunctionalTest.putObject(FunctionalTest.java:890), FunctionalTest.runObjectTests(FunctionalTest.java:3751), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... FAILED in 47 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 80,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:684:10) at endReadableNT (internal/streams/readable.js:1327:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(8/15) Running minio-py tests ... done in 19 minutes and 12 seconds
(9/15) Running s3cmd tests ... done in 2 minutes and 15 seconds
(10/15) Running s3select tests ... FAILED in 0 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(10/15) Running security tests ... done in 0 seconds

Executed 10 out of 15 tests successfully.
```
<a name="dd9b3d43-d98e-4033-a49f-7189aa963688"></a>
### 11136-55d53a2/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:31442
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 142208e6b768:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-wj0vsybpp2hndc",
    "objectName": "mbgl5blgpxwgwjbmt3gbdqftg1gx1d"
  },
  "duration": 12,
  "error": "InvalidPart: One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\n\tstatus code: 400, request id: 1651FE924303BBC0, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 1 seconds
(2/15) Running aws-sdk-php tests ... done in 41 seconds
(3/15) Running aws-sdk-ruby tests ... done in 2 seconds
(4/15) Running awscli tests ... done in 2 minutes and 1 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 33 seconds
(7/15) Running minio-dotnet tests ... done in 32 seconds
(8/15) Running minio-go tests ... done in 54 seconds
(9/15) Running minio-java tests ... done in 19 seconds
(10/15) Running minio-js tests ... done in 44 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 11 seconds
(12/15) Running s3cmd tests ... done in 14 seconds
(13/15) Running s3select tests ... FAILED in 0 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="3af16f6b-8bc5-48ce-a62b-90520b52e18f"></a>
### 11136-55d53a2/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30714
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c21d89aa1242:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-dqgpee69lg6wne",
    "objectName": "tfouwwo2rnoucfi69jr5vue4qrbp4f"
  },
  "duration": 315,
  "error": "InvalidPart: One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\n\tstatus code: 400, request id: 1651FE92981645AA, host id: ",
  "function": "testListMultipartUploads",
  "message": "AWS SDK Go CompleteMultipartUpload failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 2 seconds
(2/15) Running aws-sdk-php tests ... done in 43 seconds
(3/15) Running aws-sdk-ruby tests ... done in 4 seconds
(4/15) Running awscli tests ... done in 1 minutes and 22 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 46 seconds
(7/15) Running minio-dotnet tests ... done in 47 seconds
(8/15) Running minio-go tests ... done in 2 minutes and 23 seconds
(9/15) Running minio-java tests ... done in 1 minutes and 28 seconds
(10/15) Running minio-js tests ... done in 1 minutes and 0 seconds
(11/15) Running minio-py tests ... done in 3 minutes and 11 seconds
(12/15) Running s3cmd tests ... done in 22 seconds
(13/15) Running s3select tests ... FAILED in 1 seconds
Traceback (most recent call last):
  File "./tests.py", line 20, in <module>
    from csv import (test_csv_input_custom_quote_char,
  File "/mint/run/core/s3select/csv.py", line 22, in <module>
    from minio.selectrequest import (COMPRESSION_TYPE_NONE, FILE_HEADER_INFO_NONE,
ModuleNotFoundError: No module named 'minio.selectrequest'
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/erasure-server-sets.go
+++ b/cmd/erasure-server-sets.go
@@ -712,14 +712,21 @@ func (z *erasureServerPools) ListObjectVersions(ctx context.Context, bucket, pre
 	if err != nil && err != io.EOF {
 		return loi, err
 	}
-	loi.Objects, loi.Prefixes = merged.fileInfoVersions(bucket, prefix, delimiter, versionMarker)
-	loi.IsTruncated = err == nil && len(loi.Objects) > 0
-	if maxKeys > 0 && len(loi.Objects) > maxKeys {
-		loi.Objects = loi.Objects[:maxKeys]
+	objects := merged.fileInfoVersions(bucket, prefix, delimiter, versionMarker)
+	loi.IsTruncated = err == nil && len(objects) > 0
+	if maxKeys > 0 && len(objects) > maxKeys {
+		objects = objects[:maxKeys]
 		loi.IsTruncated = true
 	}
+	for _, obj := range objects {
+		if obj.IsDir && delimiter != "" {
+			loi.Prefixes = append(loi.Prefixes, obj.Name)
+		} else {
+			loi.Objects = append(loi.Objects, obj)
+		}
+	}
 	if loi.IsTruncated {
-		last := loi.Objects[len(loi.Objects)-1]
+		last := objects[len(objects)-1]
 		loi.NextMarker = encodeMarker(last.Name, merged.listID)
 		loi.NextVersionIDMarker = last.VersionID
 	}
@@ -728,6 +735,7 @@ func (z *erasureServerPools) ListObjectVersions(ctx context.Context, bucket, pre
 
 func (z *erasureServerPools) ListObjects(ctx context.Context, bucket, prefix, marker, delimiter string, maxKeys int) (ListObjectsInfo, error) {
 	var loi ListObjectsInfo
+
 	merged, err := z.listPath(ctx, listPathOptions{
 		Bucket:      bucket,
 		Prefix:      prefix,
@@ -741,11 +749,24 @@ func (z *erasureServerPools) ListObjects(ctx context.Context, bucket, prefix, ma
 		logger.LogIf(ctx, err)
 		return loi, err
 	}
+
 	// Default is recursive, if delimiter is set then list non recursive.
-	loi.Objects, loi.Prefixes = merged.fileInfos(bucket, prefix, delimiter)
-	loi.IsTruncated = err == nil && len(loi.Objects) > 0
+	objects := merged.fileInfos(bucket, prefix, delimiter)
+	loi.IsTruncated = err == nil && len(objects) > 0
+	if maxKeys > 0 && len(objects) > maxKeys {
+		objects = objects[:maxKeys]
+		loi.IsTruncated = true
+	}
+	for _, obj := range objects {
+		if obj.IsDir && delimiter != "" {
+			loi.Prefixes = append(loi.Prefixes, obj.Name)
+		} else {
+			loi.Objects = append(loi.Objects, obj)
+		}
+	}
 	if loi.IsTruncated {
-		loi.NextMarker = encodeMarker(loi.Objects[len(loi.Objects)-1].Name, merged.listID)
+		last := objects[len(objects)-1]
+		loi.NextMarker = encodeMarker(last.Name, merged.listID)
 	}
 	return loi, nil
 }

--- a/cmd/metacache-entries.go
+++ b/cmd/metacache-entries.go
@@ -309,7 +309,7 @@ func (m *metaCacheEntriesSorted) iterate(fn func(entry metaCacheEntry) (cont boo
 
 // fileInfoVersions converts the metadata to FileInfoVersions where possible.
 // Metadata that cannot be decoded is skipped.
-func (m *metaCacheEntriesSorted) fileInfoVersions(bucket, prefix, delimiter, afterV string) (versions []ObjectInfo, commonPrefixes []string) {
+func (m *metaCacheEntriesSorted) fileInfoVersions(bucket, prefix, delimiter, afterV string) (versions []ObjectInfo) {
 	versions = make([]ObjectInfo, 0, m.len())
 	prevPrefix := ""
 	for _, entry := range m.o {
@@ -323,7 +323,11 @@ func (m *metaCacheEntriesSorted) fileInfoVersions(bucket, prefix, delimiter, aft
 						continue
 					}
 					prevPrefix = currPrefix
-					commonPrefixes = append(commonPrefixes, currPrefix)
+					versions = append(versions, ObjectInfo{
+						IsDir:  true,
+						Bucket: bucket,
+						Name:   currPrefix,
+					})
 					continue
 				}
 			}
@@ -356,17 +360,20 @@ func (m *metaCacheEntriesSorted) fileInfoVersions(bucket, prefix, delimiter, aft
 				continue
 			}
 			prevPrefix = currPrefix
-			commonPrefixes = append(commonPrefixes, currPrefix)
-			continue
+			versions = append(versions, ObjectInfo{
+				IsDir:  true,
+				Bucket: bucket,
+				Name:   currPrefix,
+			})
 		}
 	}
 
-	return versions, commonPrefixes
+	return versions
 }
 
 // fileInfoVersions converts the metadata to FileInfoVersions where possible.
 // Metadata that cannot be decoded is skipped.
-func (m *metaCacheEntriesSorted) fileInfos(bucket, prefix, delimiter string) (objects []ObjectInfo, commonPrefixes []string) {
+func (m *metaCacheEntriesSorted) fileInfos(bucket, prefix, delimiter string) (objects []ObjectInfo) {
 	objects = make([]ObjectInfo, 0, m.len())
 	prevPrefix := ""
 	for _, entry := range m.o {
@@ -380,7 +387,11 @@ func (m *metaCacheEntriesSorted) fileInfos(bucket, prefix, delimiter string) (ob
 						continue
 					}
 					prevPrefix = currPrefix
-					commonPrefixes = append(commonPrefixes, currPrefix)
+					objects = append(objects, ObjectInfo{
+						IsDir:  true,
+						Bucket: bucket,
+						Name:   currPrefix,
+					})
 					continue
 				}
 			}
@@ -405,12 +416,15 @@ func (m *metaCacheEntriesSorted) fileInfos(bucket, prefix, delimiter string) (ob
 				continue
 			}
 			prevPrefix = currPrefix
-			commonPrefixes = append(commonPrefixes, currPrefix)
-			continue
+			objects = append(objects, ObjectInfo{
+				IsDir:  true,
+				Bucket: bucket,
+				Name:   currPrefix,
+			})
 		}
 	}
 
-	return objects, commonPrefixes
+	return objects
 }
 
 // forwardTo will truncate m so only entries that are s or after is in the list.

--- a/cmd/object-api-listobjects_test.go
+++ b/cmd/object-api-listobjects_test.go
@@ -49,6 +49,8 @@ func testListObjects(obj ObjectLayer, instanceType string, t1 TestErrHandler) {
 		"test-bucket-single-object",
 		// Listing uncommon delimiter.
 		"test-bucket-delimiter",
+		// Listing prefixes > maxKeys
+		"test-bucket-max-keys-prefixes",
 	}
 	for _, bucket := range testBuckets {
 		err := obj.MakeBucketWithLocation(context.Background(), bucket, BucketOptions{})
@@ -79,6 +81,8 @@ func testListObjects(obj ObjectLayer, instanceType string, t1 TestErrHandler) {
 		{testBuckets[3], "A/B", "contentstring", nil},
 		{testBuckets[4], "file1/receipt.json", "content", nil},
 		{testBuckets[4], "file1/guidSplunk-aaaa/file", "content", nil},
+		{testBuckets[5], "dir/day_id=2017-10-10/issue", "content", nil},
+		{testBuckets[5], "dir/day_id=2017-10-11/issue", "content", nil},
 	}
 	for _, object := range testObjects {
 		md5Bytes := md5.Sum([]byte(object.content))
@@ -467,6 +471,11 @@ func testListObjects(obj ObjectLayer, instanceType string, t1 TestErrHandler) {
 			},
 			Prefixes: []string{"file1/guidSplunk"},
 		},
+		// ListObjectsResult-36 list with nextmarker prefix and maxKeys set to 1.
+		{
+			IsTruncated: true,
+			Prefixes:    []string{"dir/day_id=2017-10-10/"},
+		},
 	}
 
 	testCases := []struct {
@@ -589,6 +598,8 @@ func testListObjects(obj ObjectLayer, instanceType string, t1 TestErrHandler) {
 		{testBuckets[4], "", "", "guidSplunk", 1000, resultCases[35], nil, true},
 		// Test listing an object with uncommon delimiter and matching prefix
 		{testBuckets[4], "file1/", "", "guidSplunk", 1000, resultCases[35], nil, true},
+		// Test listing at prefix with expected prefix markers
+		{testBuckets[5], "dir/", "", SlashSeparator, 1, resultCases[36], nil, true},
 	}
 
 	for i, testCase := range testCases {
@@ -661,26 +672,15 @@ func testListObjects(obj ObjectLayer, instanceType string, t1 TestErrHandler) {
 				}
 
 				if testCase.result.IsTruncated && result.NextMarker == "" {
-					t.Errorf("Test %d: %s: Expected NextContinuationToken to contain a string since listing is truncated, but instead found it to be empty", i+1, instanceType)
+					t.Errorf("Test %d: %s: Expected NextMarker to contain a string since listing is truncated, but instead found it to be empty", i+1, instanceType)
 				}
 
 				if !testCase.result.IsTruncated && result.NextMarker != "" {
 					if !result.IsTruncated || len(result.Objects) == 0 {
-						t.Errorf("Test %d: %s: Expected NextContinuationToken to be empty since listing is not truncated, but instead found `%v`", i+1, instanceType, result.NextMarker)
+						t.Errorf("Test %d: %s: Expected NextMarker to be empty since listing is not truncated, but instead found `%v`", i+1, instanceType, result.NextMarker)
 					}
 				}
 
-			}
-			// Take ListObject treeWalk go-routine to completion, if available in the treewalk pool.
-			for result.IsTruncated {
-				result, err = obj.ListObjects(context.Background(), testCase.bucketName,
-					testCase.prefix, result.NextMarker, testCase.delimiter, 1000)
-				if err != nil {
-					t.Fatal(err)
-				}
-				if !testCase.result.IsTruncated && len(result.Objects) > 0 {
-					t.Errorf("expected to get all objects in the previous call, but got %d more", len(result.Objects))
-				}
 			}
 		})
 	}
@@ -714,6 +714,8 @@ func testListObjectVersions(obj ObjectLayer, instanceType string, t1 TestErrHand
 		"test-bucket-single-object",
 		// Listing uncommon delimiter.
 		"test-bucket-delimiter",
+		// Listing prefixes > maxKeys
+		"test-bucket-max-keys-prefixes",
 	}
 	for _, bucket := range testBuckets {
 		err := obj.MakeBucketWithLocation(context.Background(), bucket, BucketOptions{VersioningEnabled: true})
@@ -748,7 +750,10 @@ func testListObjectVersions(obj ObjectLayer, instanceType string, t1 TestErrHand
 		{testBuckets[3], "A/B", "contentstring", nil},
 		{testBuckets[4], "file1/receipt.json", "content", nil},
 		{testBuckets[4], "file1/guidSplunk-aaaa/file", "content", nil},
+		{testBuckets[5], "dir/day_id=2017-10-10/issue", "content", nil},
+		{testBuckets[5], "dir/day_id=2017-10-11/issue", "content", nil},
 	}
+
 	for _, object := range testObjects {
 		md5Bytes := md5.Sum([]byte(object.content))
 		_, err = obj.PutObject(context.Background(), object.parentBucket, object.name, mustGetPutObjReader(t, bytes.NewBufferString(object.content),
@@ -1140,6 +1145,11 @@ func testListObjectVersions(obj ObjectLayer, instanceType string, t1 TestErrHand
 			},
 			Prefixes: []string{"file1/guidSplunk"},
 		},
+		// ListObjectsResult-36 list with nextmarker prefix and maxKeys set to 1.
+		{
+			IsTruncated: true,
+			Prefixes:    []string{"dir/day_id=2017-10-10/"},
+		},
 	}
 
 	testCases := []struct {
@@ -1262,6 +1272,8 @@ func testListObjectVersions(obj ObjectLayer, instanceType string, t1 TestErrHand
 		{testBuckets[4], "", "", "guidSplunk", 1000, resultCases[35], nil, true},
 		// Test listing an object with uncommon delimiter and matching prefix
 		{testBuckets[4], "file1/", "", "guidSplunk", 1000, resultCases[35], nil, true},
+		// Test listing at prefix with expected prefix markers
+		{testBuckets[5], "dir/", "", SlashSeparator, 1, resultCases[36], nil, true},
 	}
 
 	for i, testCase := range testCases {
@@ -1326,25 +1338,13 @@ func testListObjectVersions(obj ObjectLayer, instanceType string, t1 TestErrHand
 				}
 
 				if testCase.result.IsTruncated && result.NextMarker == "" {
-					t.Errorf("%s: Expected NextContinuationToken to contain a string since listing is truncated, but instead found it to be empty", instanceType)
+					t.Errorf("%s: Expected NextMarker to contain a string since listing is truncated, but instead found it to be empty", instanceType)
 				}
 
 				if !testCase.result.IsTruncated && result.NextMarker != "" {
 					if !result.IsTruncated || len(result.Objects) == 0 {
-						t.Errorf("%s: Expected NextContinuationToken to be empty since listing is not truncated, but instead found `%v`", instanceType, result.NextMarker)
+						t.Errorf("%s: Expected NextMarker to be empty since listing is not truncated, but instead found `%v`", instanceType, result.NextMarker)
 					}
-				}
-
-			}
-			// Take ListObject treeWalk go-routine to completion, if available in the treewalk pool.
-			for result.IsTruncated {
-				result, err = obj.ListObjectVersions(context.Background(), testCase.bucketName,
-					testCase.prefix, result.NextMarker, "", testCase.delimiter, 1000)
-				if err != nil {
-					t.Fatal(err)
-				}
-				if !testCase.result.IsTruncated && len(result.Objects) > 0 {
-					t.Errorf("expected to get all objects in the previous call, but got %d more", len(result.Objects))
 				}
 			}
 		})

--- a/cmd/web-handlers.go
+++ b/cmd/web-handlers.go
@@ -570,11 +570,13 @@ func (web *webAPIHandlers) ListObjects(r *http.Request, args *ListObjectsArgs, r
 	nextMarker := ""
 	// Fetch all the objects
 	for {
-		// Limit browser to 1000 batches to be more responsive, scrolling friendly.
-		lo, err := listObjects(ctx, args.BucketName, args.Prefix, nextMarker, SlashSeparator, 1000)
+		// Limit browser to defaults batches to be more responsive, scrolling friendly.
+		lo, err := listObjects(ctx, args.BucketName, args.Prefix, nextMarker, SlashSeparator, -1)
 		if err != nil {
 			return &json2.Error{Message: err.Error()}
 		}
+
+		nextMarker = lo.NextMarker
 		for i := range lo.Objects {
 			lo.Objects[i].Size, err = lo.Objects[i].GetActualSize()
 			if err != nil {
@@ -595,8 +597,6 @@ func (web *webAPIHandlers) ListObjects(r *http.Request, args *ListObjectsArgs, r
 				Key: prefix,
 			})
 		}
-
-		nextMarker = lo.NextMarker
 
 		// Return when there are no more objects
 		if !lo.IsTruncated {

--- a/mint/build/minio-py/install.sh
+++ b/mint/build/minio-py/install.sh
@@ -23,5 +23,5 @@ fi
 
 test_run_dir="$MINT_RUN_CORE_DIR/minio-py"
 pip3 install --user faker
-pip3 install minio==${MINIO_PY_VERSION}
+pip3 install minio=="${MINIO_PY_VERSION}"
 $WGET --output-document="$test_run_dir/tests.py" "https://raw.githubusercontent.com/minio/minio-py/${MINIO_PY_VERSION}/tests/functional/tests.py"

--- a/mint/mint.sh
+++ b/mint/mint.sh
@@ -183,7 +183,7 @@ function main()
     done
 
     ## Report when all tests in run_list are run
-    if [ $i -eq "$count" ]; then
+    if [ "$i" -eq "$count" ]; then
         echo -e "\nAll tests ran successfully"
     else
         echo -e "\nExecuted $i out of $count tests successfully."

